src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -63,6 +56,14 @@ internal class RemoteRenderer : HtmlRenderer
             _logger = logger;
         }
 
+        /// <remarks>
+        /// The RemoteRenderer may only render if we know the client is connected.
+        /// If the client disconnects during a render operation or if the underlying connection
+        /// to the client has been severed without the server knowing of it, SignalR will silently
+        /// fail the JS.RenderBatch call.
+        /// </remarks>
+        protected override bool CanRenderBatch => _client.Connected && base.CanRenderBatch;
```

Comment:
How does this work with prerendering? We need to process the render queue so that we can produce the render trees and serialize them into HTML. From what I see here, this will prevent prerendering from working.

Reply:
I guess we have to account for the offline client here which is gross. Yet another reason to figure out #7805.